ddtrace/contrib/trace_utils.py

```diff
@@ -128,6 +144,100 @@ def _get_request_header_user_agent(headers):
     return ""
 
 
+def _find_normalized_header(headers, header):
```

Comment:
We are doing basically this on several parts of the code and it's not very pretty. Eventually the right thing would be for headers to be in a key-case-sensitive dictionary subclass, which could be implemented making it store a lowered version of each key along with the real one, and on `__getitem__` lowering the argument and comparing against that one.